eng/build.yml

```diff
@@ -370,7 +370,6 @@ jobs:
       - ${{ if contains(parameters.agentOs, 'Windows_NT') }}:
         - script: eng\CIBuild.cmd
                     -configuration $(_BuildConfig)
-                    $(_PublishArgs)
```

Comment:
why this is different for Win?

Reply:
can this affect other builds?

Reply:
`$(_PublishArgs)` is just the one publishing artifacts at build step. Win already does it. To avoid publishing artifacts duplicate and sometimes it causes a race on uploading artifacts at build step of multiple jobs. In addition, our tests don't need the artifacts being published at build step.

This change is just made to the process owned by our job. It doesn't affect others.